### PR TITLE
SDT-109: Implement Additional Transformers module unit tests

### PR DESCRIPTION
**Pull request by ladamson-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SDT-109

### Change description ###
Increased transformers module unit test coverage

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: transformers/src/unit-test/java/uk/gov/moj/sdt/transformers/IndividualRequestConsumerTransformerTest.java [hunk 1]
```diff
@@ -104,10 +98,10 @@ public void testTransformJaxbToDomainAccepted() {
         transformer.transformJaxbToDomain(jaxb, domain);
 
         // Test the jaxb object has been transformed to a domain object
-        Assert.assertEquals("Request Status is incorrect", IndividualRequestStatus.ACCEPTED.getStatus(),
-                domain.getRequestStatus());
-        Assert.assertNotNull("Request updated data should be populated", domain.getUpdatedDate());
-        Assert.assertNotNull("Request completed data should be populated", domain.getCompletedDate());
+        Assertions.assertEquals(IndividualRequestStatus.ACCEPTED.getStatus(), domain.getRequestStatus(),
```
Comment (markdathornehmcts) [inline transformers/src/unit-test/java/uk/gov/moj/sdt/transformers/IndividualRequestConsumerTransformerTest.java:101]: Please add *Assertions* as an import

File: transformers/src/unit-test/java/uk/gov/moj/sdt/transformers/SubmitQueryTransformerTest.java [hunk 2]
```diff
@@ -1,129 +1,139 @@
...
+
+    /**
+     * Test the transformation from jaxb to domain object.
+     */
+    @Test
+    public void testTransformJaxbToDomain() {
+
+        // Set up the jaxb object to transform
+        final SubmitQueryRequestType jaxb = new SubmitQueryRequestType();
+        final long sdtCustomerId = 123;
```
Comment (markdathornehmcts) [inline transformers/src/unit-test/java/uk/gov/moj/sdt/transformers/SubmitQueryTransformerTest.java:80]: Query: shouldn't long be 123L?

File: transformers/src/unit-test/java/uk/gov/moj/sdt/transformers/SubmitQueryTransformerTest.java [hunk 3]
```diff
@@ -1,129 +1,139 @@
...
+import uk.gov.moj.sdt.domain.api.IBulkCustomer;
+import uk.gov.moj.sdt.domain.api.IErrorLog;
+import uk.gov.moj.sdt.domain.api.ISubmitQueryRequest;
+import uk.gov.moj.sdt.utils.AbstractSdtUnitTestBase;
+import uk.gov.moj.sdt.ws._2013.sdt.baseschema.StatusCodeType;
+import uk.gov.moj.sdt.ws._2013.sdt.submitqueryrequestschema.CriteriaType;
+import uk.gov.moj.sdt.ws._2013.sdt.submitqueryrequestschema.CriterionType;
+import uk.gov.moj.sdt.ws._2013.sdt.submitqueryrequestschema.HeaderType;
+import uk.gov.moj.sdt.ws._2013.sdt.submitqueryrequestschema.SubmitQueryRequestType;
+import uk.gov.moj.sdt.ws._2013.sdt.submitqueryresponseschema.SubmitQueryResponseType;
```
Comment (markdathornehmcts) [inline transformers/src/unit-test/java/uk/gov/moj/sdt/transformers/SubmitQueryTransformerTest.java:47]: add Assertions import here...

File: transformers/src/unit-test/java/uk/gov/moj/sdt/transformers/SubmitQueryTransformerTest.java [hunk 4]
```diff
@@ -1,129 +1,139 @@
...
+        domain.setResultCount(resultCount);
+
+        domain.setErrorLog(errorLog);
+
+        // Call the transformer
+        final SubmitQueryResponseType jaxb = transformer.transformDomainToJaxb(domain);
+
+        // Test the domain object has been transformed to a jaxb object
+        Assertions.assertEquals(sdtCustomerId, jaxb.getSdtCustomerId(), "SDT Customer ID does not match");
+        Assertions.assertEquals(resultCount, jaxb.getResultCount().longValue(), "Result count does not match");
```
Comment (markdathornehmcts) [inline transformers/src/unit-test/java/uk/gov/moj/sdt/transformers/SubmitQueryTransformerTest.java:134]: import should reduce Assertions code

File: transformers/src/unit-test/java/uk/gov/moj/sdt/transformers/SubmitQueryConsumerTransformerTest.java [hunk 5]
```diff
@@ -1,122 +1,148 @@
...
+import uk.gov.moj.sdt.domain.SubmitQueryRequest;
+import uk.gov.moj.sdt.domain.api.*;
+import uk.gov.moj.sdt.utils.AbstractSdtUnitTestBase;
+import uk.gov.moj.sdt.ws._2013.sdt.baseschema.ErrorType;
+import uk.gov.moj.sdt.ws._2013.sdt.baseschema.StatusCodeType;
+import uk.gov.moj.sdt.ws._2013.sdt.baseschema.StatusType;
+import uk.gov.moj.sdt.ws._2013.sdt.targetapp.submitqueryrequestschema.SubmitQueryRequestType;
+import uk.gov.moj.sdt.ws._2013.sdt.targetapp.submitqueryresponseschema.SubmitQueryResponseType;
+
+import static org.powermock.api.mockito.PowerMockito.when;
```
Comment (markdathornehmcts) [inline transformers/src/unit-test/java/uk/gov/moj/sdt/transformers/SubmitQueryConsumerTransformerTest.java:48]: Mockito not PowerMock

**Review by markdathornehmcts — CHANGES_REQUESTED** [review]
A few comments to peruse and resolve.

**Review by markdathornehmcts — APPROVED** [review]
Pending successful build after CVE suppressions.

**Review by markdathornehmcts — APPROVED** [review]
LGTM

File: transformers/src/unit-test/java/uk/gov/moj/sdt/transformers/BulkFeedbackTransformerTest.java [hunk 6]
```diff
@@ -70,17 +69,11 @@ class BulkFeedbackTransformerTest extends AbstractSdtUnitTestBase {
...
-            c.setAccessible(true);
-            transformer = c.newInstance();
-        } catch (final Exception e) {
-            e.printStackTrace();
-        }
-
+    public void setUpLocalTests() throws Exception {
+        // Make the constructor visible so we can get a new instance of it.
+        Constructor<BulkFeedbackTransformer> c = BulkFeedbackTransformer.class.getDeclaredConstructor();
+        c.setAccessible(true);
```
Comment (kiran-yenigala-hmcts) [inline transformers/src/unit-test/java/uk/gov/moj/sdt/transformers/BulkFeedbackTransformerTest.java:75]: why reflection is used, can't we use new BulkFeedbackTransformer()

File: transformers/src/unit-test/java/uk/gov/moj/sdt/transformers/BulkRequestTransformerTest.java [hunk 7]
```diff
@@ -67,17 +68,11 @@ class BulkRequestTransformerTest extends AbstractSdtUnitTestBase {
...
-            c = BulkRequestTransformer.class.getDeclaredConstructor();
-            c.setAccessible(true);
-            transformer = c.newInstance();
-        } catch (final Exception e) {
-            e.printStackTrace();
-        }
-
+    public void setUpLocalTests() throws Exception {
+        // Make the constructor visible so we can get a new instance of it.
+        Constructor<BulkRequestTransformer> c = BulkRequestTransformer.class.getDeclaredConstructor();
```
Comment (kiran-yenigala-hmcts) [inline transformers/src/unit-test/java/uk/gov/moj/sdt/transformers/BulkRequestTransformerTest.java:73]: plese see comment above

File: transformers/src/unit-test/java/uk/gov/moj/sdt/transformers/IndividualRequestConsumerTransformerTest.java [hunk 8]
```diff
@@ -75,17 +75,12 @@ public class IndividualRequestConsumerTransformerTest extends AbstractSdtUnitTes
...
-            c = IndividualRequestConsumerTransformer.class.getDeclaredConstructor();
-            c.setAccessible(true);
-            transformer = c.newInstance();
-        } catch (final Exception e) {
-            e.printStackTrace();
-        }
-
+    public void setUpLocalTests() throws Exception {
+        // Make the constructor visible so we can get a new instance of it.
+        Constructor<IndividualRequestConsumerTransformer> c
```
Comment (kiran-yenigala-hmcts) [inline transformers/src/unit-test/java/uk/gov/moj/sdt/transformers/IndividualRequestConsumerTransformerTest.java:80]: please see comment above

File: transformers/src/unit-test/java/uk/gov/moj/sdt/transformers/SubmitQueryConsumerTransformerTest.java [hunk 9]
```diff
@@ -1,122 +1,149 @@
...
+     */
+    private SubmitQueryConsumerTransformer transformer;
+
+    /**
+     * Set up variables for the test.
+     */
+    public void setUpLocalTests() throws Exception {
+        // Make the constructor visible so we can get a new instance of it.
+        Constructor<SubmitQueryConsumerTransformer> c
+            = SubmitQueryConsumerTransformer.class.getDeclaredConstructor();
```
Comment (kiran-yenigala-hmcts) [inline transformers/src/unit-test/java/uk/gov/moj/sdt/transformers/SubmitQueryConsumerTransformerTest.java:68]: please see above comment

File: transformers/src/unit-test/java/uk/gov/moj/sdt/transformers/UpdateItemTransformerTest.java [hunk 10]
```diff
@@ -74,16 +74,11 @@ class UpdateItemTransformerTest extends AbstractSdtUnitTestBase {
...
-            // Make the constructor visible so we can get a new instance of it.
-            c = UpdateItemTransformer.class.getDeclaredConstructor();
-            c.setAccessible(true);
-            updateItemTransformer = c.newInstance();
-        } catch (final Exception e) {
-            LOGGER.error("Error during construction of the UpdateItemTransformer", e);
-        }
+    public void setUpLocalTests() throws Exception {
+        // Make the constructor visible so we can get a new instance of it.
+        Constructor<UpdateItemTransformer> c = UpdateItemTransformer.class.getDeclaredConstructor();
```
Comment (kiran-yenigala-hmcts) [inline transformers/src/unit-test/java/uk/gov/moj/sdt/transformers/UpdateItemTransformerTest.java:79]: please see above comment

**Review by hemantt — APPROVED** [review]

**Review by markdathornehmcts — APPROVED** [review]
LGTM

**Pull request merged**